File: greedy.c
#include <cs50.h>
#include <stdio.h>

int main()
{
    printf("O hai!");
    float money = -1.0;
    while(money < 0)
    
    {
    printf("How much change is owed?\n");
    money = get_float();
        
    }
    
     int money_1=money*100+0.5;
   
     int money_25 = money_1/25;
     //printf("%i\n",money_25);
     
     int money_ref =money_1%25;
     //printf("%i\n", money_ref);
     
     int money_10 = money_ref/10;
     int money_ref10 =money_ref%10;
     
     int money_5 = money_ref10/5;
     int money_ref5 =money_ref10%5;
     
     int money1 = money_ref5;
     
     int money_s = money_25 + money_10 + money_5 + money1;
     printf ("%i\n",money_s);
     
}

File: greedy.py
#!usr/bin/python3

import sys
import cs50

def main():
	print("O hai! ", end="")
	while True:
		print("How much change is owned?: ", end="")
		money = cs50.get_float()
		if money >= 0:
			break

	money_in_cents = int(money*100)
	avaliable_coins = [25, 10, 5, 1]
	total_coins = 0
	for coin in avaliable_coins:
		coins_to_add = money_in_cents//coin
		total_coins += coins_to_add
		if money_in_cents % coin == 0:
			break

		money_in_cents -= coins_to_add * coin

	print(total_coins)
	return 0

if __name__ == "__main__":
	ret = main()
	exit(ret)